pages/api/auth.ts

```diff
@@ -1,13 +1,42 @@
 import { NextApiRequest, NextApiResponse } from 'next';
+import { createAppExtension } from '@lib/appExtensions';
 import { encodePayload, getBCAuth, setSession } from '../../lib/auth';
 
 export default async function auth(req: NextApiRequest, res: NextApiResponse) {
     try {
         // Authenticate the app on install
         const session = await getBCAuth(req.query);
         const encodedContext = encodePayload(session); // Signed JWT to validate/ prevent tampering
+        const { access_token: accessToken, context } = session;
+
+        const storeHash = context?.split('/')[1] || '';
 
         await setSession(session);
+
+        /**
+         * For stores that do not have the app installed yet, create App Extensions when app is
+         * installed.
+         */
+
+        const requestBody = await createAppExtension();
+
+        const response = await fetch(
+            `https://${process.env.API_URL}/stores/${storeHash}/graphql`,
+            {
+                method: 'POST',
+                headers: {
+                    accept: 'application/json',
+                    'content-type': 'application/json',
+                    'x-auth-token': accessToken,
+                },
+                body: JSON.stringify(requestBody),
+            }
+        );
+        const errors = await response.json();
+        if (errors.length > 0) {
+            throw new Error(errors[0]?.message);
+        }
```

Comment:
`await response.json()` returns a resolved promise of the body from the POST request to create an app extension. Naming the variable as `errors` is misleading, because the body could be an error response, _or a success response_. 

If your goal here is to check if the GraphQL response returned an error, I think the better way to go about this would be using the [`Response: ok` property](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok)

```ts
if (!response.ok) {
  const error = await response.json();
  throw new Error(error.message); // keep in mind, error.message might be undefined if the `response.json()` object does not have a top-level message property.  
}
```

Reply:
Updated and added in a default string to throw if the message is undefined